restapi.go

```diff
@@ -1268,6 +1268,109 @@ func (s *Session) GuildEmojiDelete(guildID, emojiID string) (err error) {
 	return
 }
 
+// GuildTemplate returns a GuildTemplate for the given code
+// templateCode: The Code of a GuildTemplate
+func (s *Session) GuildTemplate(templateCode string) (st *GuildTemplate, err error) {
+
+	body, err := s.RequestWithBucketID("GET", EndpointGuildTemplate(templateCode), nil, EndpointGuildTemplate(templateCode))
+	if err != nil {
+		return
+	}
+
+	err = unmarshal(body, &st)
+	return
+}
+
+// GuildCreateWithTemplate creates a guild based on a GuildTemplate
+// templateCode: The Code of a GuildTemplate
+// name: The name of the guild (2-100) characters
+func (s *Session) GuildCreateWithTemplate(templateCode, name string) (st *Guild, err error) {
+
+	data := struct {
+		Name string `json:"name"`
```

Comment:
It seems you forgot to include `icon`.
Referring to [the docs](https://discord.com/developers/docs/resources/guild-template#create-guild-from-guild-template)